scripts/generate-upgrade-dashboard.js

```diff
@@ -0,0 +1,196 @@
+  );
+  if (!response.ok) throw new Error(`HTTP error! status: ${response.status}`);
+
+  const releases = await response.json();
+  const stableRelease = releases.find(
+    release =>
+      !release.prerelease &&
+      !release.tag_name.includes('next') &&
+      !release.tag_name.includes('rc'),
+  );
+
+  if (!stableRelease) throw new Error('No stable releases found');
+  return stableRelease.tag_name;
+}
+
+// get all workspace versions
+function getWorkspaceVersions() {
+  const workspacesDir = path.join(__dirname, '..', 'workspaces');
+  const workspaces = [];
+
+  if (!fs.existsSync(workspacesDir)) {
+    throw new Error('Workspaces directory not found');
+  }
+
+  const entries = fs.readdirSync(workspacesDir, { withFileTypes: true });
+
+  for (const entry of entries) {
+    if (entry.isDirectory()) {
+      const backstageJsonPath = path.join(
+        workspacesDir,
+        entry.name,
+        'backstage.json',
+      );
+
+      if (fs.existsSync(backstageJsonPath)) {
+        try {
+          const backstageJson = JSON.parse(
+            fs.readFileSync(backstageJsonPath, 'utf8'),
+          );
+          if (backstageJson.version) {
+            workspaces.push({
+              name: entry.name,
+              version: backstageJson.version,
+            });
+          }
+        } catch (error) {
+          console.warn(
+            `Warning: Could not read version from ${entry.name}/backstage.json:`,
+            error.message,
+          );
+        }
+      }
+    }
+  }
+
+  return workspaces;
+}
+
+// calculate version difference
+function getVersionDifference(currentVersion, latestVersion) {
+  const current = semver.clean(currentVersion);
+  const latest = semver.clean(latestVersion);
+
+  if (!current || !latest) return 0;
+
+  // if major versions differ, treat as significantly outdated
+  if (semver.major(current) !== semver.major(latest)) {
+    return semver.major(latest) > semver.major(current) ? 10 : 0;
+  }
+
+  return semver.minor(latest) - semver.minor(current);
+}
+
+// categorize workspaces by how outdated they are
+function categorizeWorkspaces(workspaces, latestVersion) {
+  const tiers = { tier1: [], tier2: [], tier3: [] };
+
+  workspaces.forEach(workspace => {
+    const versionDiff = getVersionDifference(workspace.version, latestVersion);
+    if (versionDiff >= 3) tiers.tier1.push(workspace);
+    else if (versionDiff === 2) tiers.tier2.push(workspace);
+    else if (versionDiff === 1) tiers.tier3.push(workspace);
+  });
+
+  // sort each tier by workspace name
+  Object.values(tiers).forEach(tier =>
+    tier.sort((a, b) => a.name.localeCompare(b.name)),
+  );
+
+  return tiers;
+}
+
+// generate markdown table for a tier
+function generateTierTable(workspaces, emoji, title, description) {
+  if (workspaces.length === 0) return '';
+
+  let output = `## ${emoji} ${title}${
+    description ? ` – ${description}` : ''
+  }\n\n`;
+  output += '| Workspace | Current Version |\n';
```

Comment:
Would it be possible to include the owner(s)? That could be helpful.

Reply:
As this isn't blocking I'd like to land and iterate 🙏🏻  I am open to adding codeowners in a follow-up!